lib/galaxy/tools/__init__.py

```diff
@@ -3137,9 +3138,9 @@ def add_copied_value_to_new_elements(new_label, dce_object):
             if new_label in new_elements:
                 raise Exception(f"New identifier [{new_label}] appears twice in resulting collection, these values must be unique.")
             if getattr(dce_object, "history_content_type", None) == "dataset":
-                copied_value = dce_object.copy(flush=False)
+                copied_value = dce_object.copy(copy_tags=dce_object.tags, flush=False)
             else:
-                copied_value = dce_object.copy()
+                copied_value = dce_object.copy(copy_tags=dce_object.tags)
```

Comment:
Sorry, missed this one.
```suggestion
                copied_value = dce_object.copy()
```